ts/import-csv/index.ts

```diff
@@ -32,22 +26,15 @@ const i18n = setupI18n({
 });
 
 export async function setupImportCsvPage(path: string): Promise<ImportCsvPage> {
-    const gettingMetadata = getCsvMetadata(path);
-
-    let notetypes: Notetypes.NotetypeNames;
-    let decks: Decks.DeckNames;
-    let metadata: ImportExport.CsvMetadata;
-    try {
-        [notetypes, decks, metadata] = await Promise.all([
-            gettingNotetypes,
-            gettingDecks,
-            gettingMetadata,
-            i18n,
-        ]);
-    } catch (err) {
-        alert(err);
-        throw (err);
-    }
```

Comment:
I think the alert didn't work anyway. Was that why you've removed it? Handling this more gracefully is still on the todo list.

Reply:
The RPC methods automatically alert when an error occurs by default, and processing should stop after one of the promises fails, so I think we can probably get away without an outer catch.

Reply:
If we're talking about a popup dialog here, it doesn't work for me. If the file can't be parsed, there's nothing but an empty window.

Reply:
Do you mean on this branch? Not sure what's happening there; I was getting a pop-up in my testing. I've fixed it so instead of a generic error it relays the backend message now. Tested on Windows too to make sure it works there:
![2023-06-14T22:14:09,805846314+10:00](https://github.com/ankitects/anki/assets/180542/8c5f587b-29bc-486a-a2e4-8e6f1d89ebb0)

Reply:
I see, you're testing with invalid Unicode. That works here, too. I was testing with an empty file, but that seems to be a special case, because it's a frontend error.
The reason is the slicing in ImportCsvPage.svelte, line 67. But probably the dialog should abort earlier for empty files.

Reply:
Fixed :-) https://github.com/ankitects/anki/commit/de8f62f831402a997f4eba5f9a69b403ee2f004d